.github/workflows/ci-build.yaml

```diff
@@ -23,8 +23,25 @@ permissions:
   contents: read
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      backend: ${{ steps.filter.outputs.backend }}
+      frontend: ${{ steps.filter.outputs.frontend }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
```

Comment:
I used [`tj-actions/changed-files`](https://github.com/tj-actions/changed-files), but I suppose either works.

You will want to change the versions to use hashes with a version comment, like the other actions in the repo (for supply chain security purposes a la #11359 / #11360, as GH action versions are mutable)

Reply:
Good catch @agilgur5, fixed!

Reply:
For posterity, this was changed to `tj-actions/changed-files` in #17180